server/datastore/mysql/hosts_test.go

```diff
@@ -3407,9 +3407,41 @@ func testHostsListMacOSSettingsDiskEncryptionStatus(t *testing.T, ds *Datastore)
 	// verifying status
 	upsertHostCPs([]*fleet.Host{hosts[0], hosts[1]}, []*fleet.MDMAppleConfigProfile{noTeamFVProfile}, fleet.MDMOperationTypeInstall, &fleet.MDMDeliveryVerifying, ctx, ds, t)
 	oneMinuteAfterThreshold := time.Now().Add(+1 * time.Minute)
-	createDiskEncryptionRecord(ctx, ds, t, hosts[0].ID, "key-1", true, oneMinuteAfterThreshold)
+	// host 0 needs to finish key rotation (action required), host 1 has finished key rotation but profile is verifying
```

Comment:
> host 1 has finished key rotation but profile is verifying

This scenario happens if the escrow key is ingested before the profile is confirmed to be applied by osquery?

Reply:
Yeah. I figured this was due to MDM reporting on this but it looks like we're ingesting both things via osquery, so this is a corner case. This test is mainly there to exercise all potential permutations hit by the DB query, and I'm assuming the DB query has a bucket for this for a reason.